modules/Makefile.docker-machine

```diff
@@ -0,0 +1,54 @@
+DOCKER_MACHINE_NAME ?= gladly
+DOCKER_DOWNLOAD_URL ?= https://get.docker.com/builds/$(LOCAL_OS)/$(LOCAL_ARCH)/docker-$(DOCKER_VERSION)
+DOCKER_ISO ?= https://github.com/boot2docker/boot2docker/releases/download/v$(DOCKER_VERSION)/boot2docker.iso
+
+DEPS_TARGETS += machine:deps
+
+.PHONY : machine\:deps machine\:info machine\:recreate machine\:create machine\:destroy machine\:start machine\:stop machine\:restart machine\:deps
+
+## Setup dependencies for docker-machine
+machine\:deps:
+	@[ -x $(DOCKER_CMD) ] || (curl -s -o $(DOCKER_CMD) $(DOCKER_DOWNLOAD_URL) && chmod 755 $(DOCKER_CMD))
+	@(docker-machine status $(DOCKER_MACHINE_NAME) 2>&1 | grep -v -q "Host does not exist") || $(SELF) machine:create
+	@(docker-machine status $(DOCKER_MACHINE_NAME) 2>&1 | grep -q "^Running") || $(SELF) machine:start
+	@docker-machine env $(DOCKER_MACHINE_NAME) | tr -d '"' > "$(MAKEFILE_PATH).env"
+	$(eval -include $(MAKEFILE_PATH).env)
+	@$(call assert,DOCKER_HOST)
+	@$(call assert,DOCKER_CERT_PATH)
+
+## Docker Machine Info
+machine\:info:
+	@echo 'DOCKER_MACHINE_NAME=$(DOCKER_MACHINE_NAME)'
+	@echo 'DOCKER_DOWNLOAD_URL=$(DOCKER_DOWNLOAD_URL)'
+	@echo 'DOCKER_HOST=$(DOCKER_HOST)'
+	@echo 'DOCKER_CERT_PATH=$(DOCKER_CERT_PATH)'
+
+## Recreate the docker machine
+machine\:recreate: destroy-machine create-machine
+
+## Create docker machine for environment
+machine\:create:
```

Comment:
I think it makes more sense to put it in this repo than docker-compose to reduce the levels of indirection for production code. I also think it would make more sense logically for this to be in a separate shared repo, but that is too much overhead to justify. Maybe we should make a mono repo.